pkg/controller/statusmanager/status_manager.go

```diff
@@ -203,12 +320,80 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...operv
 		log.Printf("Failed to set operator status: %v", err)
 	}
 
+	if status.hyperShiftConfig.Enabled {
```

Comment:
Can you add a comment indicating what this is doing? (Namely, mirroring in to the HostedControlPlane type). And consider breaking it out in to its own function -- `WriteHypershiftStatus()` or something like that.

Reply:
will update

Reply:
Updated